`std::unexpected()` deprecated

The function  `epicsThread :: printLastChanceExceptionMessage` in modules/libcom/src/osi/epicsThread.cpp calls `std::unexpected ()`.
https://github.com/epics-base/epics-base/blob/fa00572780a00bea2f57dbff69653b14bd792be0/modules/libcom/src/osi/epicsThread.cpp#L84

The gcc-12 compiler claims that `std::unexpected ()` is deprecated:
```
../osi/epicsThread.cpp: In member function ‘void epicsThread::printLastChanceExceptionMessage(const char*, const char*)’:
../osi/epicsThread.cpp:84:21: warning: ‘void std::unexpected()’ is deprecated [-Wdeprecated-declarations]
   84 |     std::unexpected ();
      |     ~~~~~~~~~~~~~~~~^~
In file included from ../osi/epicsThread.cpp:14:
/opt/rh/gcc-toolset-12/root/usr/include/c++/12/exception:109:8: note: declared here
  109 |   void unexpected() __attribute__ ((__noreturn__));
      |        ^~~~~~~~~~
```
When using `-std=c++23`, compilation fails:
```
../osi/epicsThread.cpp: In member function ‘void epicsThread::printLastChanceExceptionMessage(const char*, const char*)’:
../osi/epicsThread.cpp:84:10: error: ‘unexpected’ is not a member of ‘std’
   84 |     std::unexpected ();
      |          ^~~~~~~~~~
```
See also: https://en.cppreference.com/w/cpp/error/unexpected

--- a/modules/libcom/src/osi/epicsThread.cpp
+++ b/modules/libcom/src/osi/epicsThread.cpp
@@ -16,6 +16,7 @@
 
 #include <stdio.h>
 #include <stddef.h>
+#include <stdlib.h>
 #include <float.h>
 #include <string.h>
 
@@ -78,10 +79,8 @@ void epicsThread :: printLastChanceExceptionMessage (
         "with type \"%s\" in thread \"%s\" at %s\n",
         pExceptionContext, pExceptionTypeName, name, date );
     errlogFlush ();
-    // This behavior matches the C++ implementation when an exception
-    // isn't handled by the thread code. Users can install their own
-    // application-specific unexpected handler if preferred.
-    std::unexpected ();
+
+    abort();
 }
 
 extern "C" void epicsThreadCallEntryPoint ( void * pPvt )

